### ISSUE TITLE
test-nbd race condition hitting some automated tests causing a false-positive and a 30 minute hang

### ISSUE THREAD

**Issue by cpaelzer** [systemd/casync#113]
Hi,
I happend to find that e.g. on most automated tests in Ubuntu there was a fail on test-nbd like [1]
This was independent to architecture and seemed to hit ~90% of the cases.

OTOH when one ran the tests locally I couldn't reproduce in 5/5 cases.
After a while I found that in the minimal environment that the autopkgtests (under kvm) there is a race.
The target that is made available is not yet ready when the subsequent dd tries to access it.

```
++ /tmp/autopkgtest.c37m28/build.c4y/casync-2/obj-x86_64-linux-gnu/notify-wait /tmp/autopkgtest.c37m28/build.c4y/casync-2/obj-x86_64-linux-gnu/c
  + MKDEV_PID=7357
  + /tmp/autopkgtest.c37m28/build.c4y/casync-2/obj-x86_64-linux-gnu/test-calc-digest sha512-256
  + dd if=/var/tmp/test-casync.rnd/test-node bs=102400 count=80
  0+0 records in
  0+0 records out
  0 bytes copied, 1,8723e-05 s, 0,0 kB/s
```

Furthermore this not only breaks the test to be bad (checksum doesn't match).
On top it causes the test to become a zombified process and then hang for ~30 minutes until meson gives up on it.

`Zs ? 0:00 \_ [test-nbd.sh] <defunct>`

A first try with a trivial sleep 5 solved all my cases, but I thought that might be a useless extra wait for fast CI runs.
So I implemented a simple retry with an initial sleep of 0s being a no-op in all non-affected cases.
But providing up to !5 sec race window for cases that are affected - which should be enough.

I'll provide a PR in a few minutes with a suggested fix.

[1]: https://objectstorage.prodstack4-5.canonical.com/v1/AUTH_77e2ada1e7a84929a74ba3b87153c0ac/autopkgtest-bionic/bionic/s390x/c/casync/20171206_111728_2644b@/log.gz

**Comment by cpaelzer** [comment]
#114

**Comment by cpaelzer** [comment]
Will be fixed by #118 once released.

**Comment by poettering** [comment]
Let's close this as #118 got merged.
